src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2039,6 +2072,20 @@ theorem sub_eq_xor {a b : BitVec 1} : a - b = a ^^^ b := by
   have hb : b = 0 ∨ b = 1 := eq_zero_or_eq_one _
   rcases ha with h | h <;> (rcases hb with h' | h' <;> (simp [h, h']))
 
+/-! ### smtSDiv -/
+
+theorem smtSDiv_eq {x y : BitVec n} : x.smtSDiv y =
+  match x.msb, y.msb with
+  | false, false => smtUDiv x y
+  | false, true  => .neg (smtUDiv x (.neg y))
+  | true,  false => .neg (smtUDiv (.neg x) y)
+  | true,  true  => smtUDiv (.neg x) (.neg y) := rfl
+
+@[simp]
+theorem smtSDiv_zero {x : BitVec n} : x.smtSDiv 0#n = if x.slt 0#n then 1 else -1 := by
+  simp [smtSDiv]
```

Comment:
```suggestion
theorem smtSDiv_zero {x : BitVec n} : x.smtSDiv 0#n = if x.slt 0#n then 1#w else (allOnes w) := by
  simp [smtSDiv]
```

Should we use `1#w` as simp normal form and `allOnes w` to model all ones?

Reply:
Done.